src/analysis/statistical_analysis/high_water_level_analyzer.py

```diff
@@ -0,0 +1,64 @@
+from src.analysis.statistical_analysis.stat_calculator import StatCalculator
+from src.graph_building.interfaces.vertex_data_interface import VertexDataInterface
+from src.graph_manipulation.flood_wave_filter import FloodWaveFilter
+
+
+class HighWaterLevelAnalyzer:
+    """
+    This class gets statistics of high water level (red) flood waves.
+    """
+    @staticmethod
+    def get_red_wave_count_at_station(flood_waves: list,
```

Comment:
Mivel kb. ugyanazokat az argumentumokat várja mindkét metódus, lehetne egy normális, nem statikus osztályunk, ahol a konstruktorban kérjük be ezeket és a metódusoknak már kb. nincs bemeneti argumentuma. (A `statistic` az kell az aggregáló metódushoz.)